virtcontainers/factory/factory.go

```diff
@@ -63,7 +63,7 @@ func NewFactory(ctx context.Context, config Config, fetchOnly bool) (vc.Factory,
 				return nil, err
 			}
 		} else {
-			b = template.New(ctx, config.VMConfig)
+			b, err = template.New(ctx, config.VMConfig)
```

Comment:
Seems the error value not used at all, IMHO, we need more thoughts on error processing of factory initialization, maybe @bergwolf could give more comments.

Reply:
Yeah, please return `err` here.

@liwei Are you referring to the fact that we currently fall back to direct vm when vm templating is unsupported?

Reply:
yep, I have no strong favorite on which(fallback or error out) is better, my concern is that silently fallback may hide some information to users and make them think factory works as they expected.

Reply:
hmm, I agree. Do you want to make a PR for it?

Reply:
NVM, it seems such a change should be folded into this PR instead.